### DISCUSSION TITLE
اجرای دستور ری استارت مرزبان بصورت دوره ای

### DISCUSSION THREAD

**Discussion by Lordrichard23** [Gozargah/Marzban#711] [Q&A]
سلام دوستان
من به علت مشکل سرور پنل مرزبانم بصورت دوره ای متوقف میشه برای جلوگیری از این مشکل یا باید سرور رو عوض کنم یا بصورت دوره ای مثلا کرونجابی تنظیم کنم که هر ۸ ساعت اتومات مرزبان ری استارت بشه ، که فعلا برام مقدور نیست تعویض اون

نحوه تنظیم زمان رو بلدم در کرونجاب اما نمیدونم دقیقا برای ری استارت مرزبان چه کدی بنویسم 

اگر اطلاعی دارید ممنون میشم راهنمایی کنید

**Comment by omid-sadeghi** [comment]
اگر شما داخل کرونجاب بنویسید `marzban restart` کار نمیکنه. باید مسیر کامل دستور  `marzban`رو بهش بدید. برای اینکه مسیر کامل فایل `marzban` رو پیدا کنی، از دستور `whereis marzban` استفاده کن. مثلا برای من توی این مسیر هست: `usr/local/bin/marzban/`

پس دستوری که برای من کار می‌کنه میشه:
`usr/local/bin/marzban restart/`
اگر برای شما هم `marzban` توی همین مسیر باشه همین کار میکنه وگرنه مسیر متفاوت رو مینویسی.

اتفاقا ایده‌ی ریستارت مرزبان از طریق کرون فکر خیلی خوبی هست. خودم هم انجامش دادم چون برای من هم مرزبان بعد یکی دو روز یا کمتر کند میشه و باید دستی ریستارتش میکردم.

**Comment by m0x61h0x64i** [comment]
قطعی داره، یه راه نیست فقط xray ریستارت شه؟
@M03ED

**Comment by waxilu** [comment]
متاسفانه بعد از اپدیت به نسخه ۴ و بعد از ۴ این باگ برای خیلی ها پیش اومده

کافیه دستور زیر رو ابتدا اجرا کنید

`crontab -e`

سپس دستور زیر داخلش قرار بدید

`1 1 * * * marzban restart`

این دستور ری استارت مرزبان ساعت ۱ شب اتفاق میفته یک بار نصف شب انجام بشه خیلی بهتره

به نظرم هر ۸ ساعت نزارید چون یوزرها دچار قطی میشن.

**Reply by kamiepic** [reply]
آیا همین کافیه؟  `1 1 * * * marzban restart`
یا باید لوکیشنم بدیم مثل این
`1 1 * * * /usr/local/bin/marzban restart`
من خودم بدون لوکیشنش دادم حالا نمیدونم کار میکنه یا نه

**Reply by waxilu** [reply]
بله کافیه اجرا میشه

**Comment by omid-sadeghi** [comment]
> آیا همین کافیه؟ `1 1 * * * marzban restart` یا باید لوکیشنم بدیم مثل این `1 1 * * * /usr/local/bin/marzban restart` من خودم بدون لوکیشنش دادم حالا نمیدونم کار میکنه یا نه

خودم به شخصه هر دستوری که توی کرون اجرا بخام بکنم، تا مسیر کامل ندم اجرا نمیشه. اما هر سیستمی مطمئنا تنظیماتش فرق میکنه. شما کافیه بعد از دستور کرون خروجی رو بفرستی روی یه فایل متنی. اینطوری میتونی چک کنی که واقعا اجرا شده یا نه. مثل دستور زیر:

`marzban restart >> /root/log_marzban_restart.txt`

با دستور بالا خروجی دستور `marzban restart` داخل فایل `log_marzban_restart.txt` که در دایرکتوری `root` هست نوشته میشه. اگر بعد از اون ساعتی که قراره کرون اجرا بشه داخل فایلی متنی خالی بود، یعنی که اجرا نشده. اگر داخلش خروجی لاگ مرزبان بود که سعنی اجرا شده.
